var/spack/repos/builtin/packages/dpcpp/package.py

```diff
@@ -101,45 +144,59 @@ def cmake_args(self):
             self.define("XPTI_SOURCE_DIR", xpti_dir),
             self.define("LLVM_EXTERNAL_XPTIFW_SOURCE_DIR", xptifw_dir),
             self.define("LLVM_EXTERNAL_LIBDEVICE_SOURCE_DIR", libdevice_dir),
+            self.define("LLVM_EXTERNAL_SYCL_FUSION_SOURCE_DIR", fusion_dir),
             self.define("LLVM_ENABLE_PROJECTS", llvm_enable_projects),
             self.define("LIBCLC_TARGETS_TO_BUILD", libclc_targets_to_build),
-            self.define_from_variant("SYCL_BUILD_PI_CUDA", "cuda"),
-            self.define_from_variant("SYCL_BUILD_PI_ROCM", "rocm"),
```

Comment:
Are all these changes compatible with older DPC++ versions? I'm guessing not: you should find what version the internals changed from using ROCM to HIP (for example) and add logic here so that older versions will still compile. Then when the "deprecated" versions are removed you can delete them.